tiledb/array.py

```diff
@@ -163,6 +148,21 @@ def index_domain_subarray(array, dom, idx: tuple):
 
         start, stop, step = dim_slice.start, dim_slice.stop, dim_slice.step
 
+        # In the case that current domain is non-empty, we need to consider it
+        if (
+            start is None
+            and stop is None
+            and hasattr(array.schema, "current_domain")
+            and not array.schema.current_domain.is_empty
+        ):
+            subarray.append(
+                (
+                    array.schema.current_domain.ndrectangle.range(r)[0],
+                    array.schema.current_domain.ndrectangle.range(r)[1],
+                )
+            )
+            continue
+
```

Comment:
How does this interact with the `if` statement below? Is there any possibility of adding a subarray twice?

Reply:
I don't see a way for a subarray to be added twice for the same dimension. The `continue` statement will prevent this from happening if both `start` and `stop` are `None` and there is a non-empty current domain.

Reply:
Nevermind, I overlooked the continue - thanks!